src/lib/draw/route/RouteControls.svelte

```diff
@@ -11,53 +11,36 @@
     avoid_doubling_back: $userSettings.avoidDoublingBack,
     extend_route: extendRoute,
   });
-
-  // TODO It'd be nice to wire up reactivity to RouteTool directly.
-  let snapping = true;
-  function onKeyDown(e: KeyboardEvent) {
-    if (e.key == "Shift") {
-      // Don't stop propagation, so RouteTool can see this
-      snapping = false;
-    }
-  }
-  function onKeyUp(e: KeyboardEvent) {
-    if (e.key == "Shift") {
-      // Don't stop propagation, so RouteTool can see this
-      snapping = true;
-    }
-  }
 </script>
 
-<svelte:window on:keydown={onKeyDown} on:keyup={onKeyUp} />
-
-{#if snapping}
-  <p>
-    Snapping to existing roads. Press <b>Shift</b>
+{#if $routeToolSnapMode}
+  <p style="background: red; padding 8px;">
+    Snapping to existing roads. Press <b>s</b>
     to draw anywhere
   </p>
 {:else}
-  <p>
-    Drawing points anywhere. Release <b>Shift</b>
+  <p style="background: blue; padding: 8px;">
+    Drawing points anywhere. Release <b>s</b>
```

Comment:
Two small notes: it's really `press s to switch back to snapped mode` since you're not holding s down for this.

The other one I'm a little less certain of but the behaviour when you switch back to snapped isn't entirely obvious (although I can't think it would better in the more intuitive way). Difficult to explain in text maybe have a quick call in a mo?

Reply:
Fixed the text and coloring lines now to be very clear